packages/flutter_tools/lib/src/commands/analyze_once.dart

```diff
@@ -45,38 +46,62 @@ class AnalyzeOnce extends AnalyzeBase {
       }
     }
 
-    final bool currentDirectory = argResults['current-directory'] && (argResults.wasParsed('current-directory') || dartFiles.isEmpty);
     final bool currentPackage = argResults['current-package'] && (argResults.wasParsed('current-package') || dartFiles.isEmpty);
     final bool flutterRepo = argResults['flutter-repo'] || inRepo(argResults.rest);
 
-    //TODO (pq): revisit package and directory defaults
-
-    if (currentDirectory  && !flutterRepo) {
-      // ./*.dart
-      final Directory currentDirectory = fs.directory('.');
-      bool foundOne = false;
-      for (FileSystemEntity entry in currentDirectory.listSync()) {
-        if (isDartFile(entry)) {
-          dartFiles.add(entry);
-          foundOne = true;
+    // Use dartanalyzer directly except when analyzing the Flutter repository.
+    // Analyzing the repository requires a more complex report than dartanalyzer
+    // currently supports (e.g. missing member dartdoc summary).
+    // TODO(danrubel) enhance dartanalyzer to provide this type of summary
+    if (!flutterRepo) {
+      final List<String> arguments = <String>[];
+      arguments.addAll(dartFiles.map((FileSystemEntity f) => f.path));
+
+      if (arguments.length < 1 || currentPackage) {
```

Comment:
`arguments.length < 1` is better written `arguments.isEmpty`
Amusingly this is a creative-enough way to write `isEmpty` that the lint that looks for this kind of thing didn't catch it. :-)